src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -61,7 +64,11 @@ internal static MsalServiceException FromHttpResponse(
                     innerException);
             }
 
-            ex ??= new MsalServiceException(errorCode, errorMessage, innerException);
+            ex ??= new MsalServiceException(errorCode,
+                httpResponse.StatusCode == HttpStatusCode.NotFound
+                        ? errorMessage + " Authority used: " + context.ApiEvent.TokenEndpoint.Split('?')[0]
```

Comment:
1. Why don't you log the entire token endpoint?
2. Possible Null ref expections? HJow do you know "TokenEndpoint" is not null?

Reply:
Im using the wrong value here, token endpoint instead of authority. I think I misread the code somewhere. For the authority, the requirement is to not have the query params added which is why I am removing them here.